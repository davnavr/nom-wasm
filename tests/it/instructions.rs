use nom_wasm::{error::VerboseError, isa::instructions};
use std::fmt::Write;

#[test]
fn basic_expr() {
    let expr: &[u8] = &[0x20, 0x00, 0x41, 0x2A, 0x6A, 0x0F, 0x01, 0x0B];
    let mut results = allocator_api2::vec::Vec::with_capacity(6);

    nom_wasm::isa::expr::<_, VerboseError>(expr, instructions::Parser::new(&mut results)).unwrap();

    let mut text = String::with_capacity(128);
    for instr in results.into_iter() {
        let _ = writeln!(&mut text, "{instr}");
    }

    insta::assert_display_snapshot!(&text);
}
