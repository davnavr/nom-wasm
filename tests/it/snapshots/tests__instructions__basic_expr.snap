---
source: tests/it/instructions.rs
expression: "&text"
---
local.get 0
i32.const 0x0000002A (* signed = 42, unsigned = 42 *)
i32.add
return
nop
end

